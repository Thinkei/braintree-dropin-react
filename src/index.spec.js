/* eslint-disable */
import 'babel-polyfill'
import React from 'react'
import { shallow, mount } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import braintree from 'braintree-web-drop-in'
import BraintreeDropin from '../dist'

describe('<BraintreeDropin />', () => {
  it('should render', () => {
    const component = shallow(<BraintreeDropin braintree={braintree} authorizationToken='client_token' handlePaymentMethod={() => {}} submitButtonText='TEXT' />)
    const tree = shallowToJson(component)
    expect(tree).toMatchSnapshot()
  });

  it('should have a submit button', function () {
    const component = shallow(<BraintreeDropin braintree={braintree} authorizationToken='client_token' handlePaymentMethod={() => {}} submitButtonText='TEXT' />)
    expect(component.find('button')).toHaveLength(1);
  })

  it('should render with custom button text', function () {
    const component = mount(<BraintreeDropin braintree={braintree} authorizationToken='client_token' handlePaymentMethod={() => {}} submitButtonText='TEXT' />)
    expect(component.prop('submitButtonText')).toEqual('TEXT');
  })
})
