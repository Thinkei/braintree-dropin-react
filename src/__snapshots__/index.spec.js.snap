// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BraintreeDropin /> should render 1`] = `
<div
  className="braintree-dropin-react"
>
  <div
    className="braintree-dropin-react-form"
  />
  <div
    className="braintree-dropin-react-submit-btn-wrapper"
  >
    <button
      disabled={true}
      onClick={[Function]}
    >
      TEXT
    </button>
  </div>
</div>
`;
